common/src/main/java/io/druid/metadata/PasswordProvider.java

```diff
@@ -34,5 +34,15 @@
 })
 public interface PasswordProvider
```

Comment:
I was thinking `SecretProvider` would look like the new version of `PasswordProvider` that you have in this PR. For this PR, I think it'd make sense to just do that and not touch `PasswordProvider` or metadata storage at all; then do a later PR to unify them.